transcode.py

```diff
@@ -1,22 +1,47 @@
+from dataclasses import dataclass
 import logging
 import os
 from typing import Optional
+import time
 
 import base_util
 from config import data_base_dir
 
 logger = logging.getLogger(__name__)
 
 
-def try_transcode(input_path, asset_id, extension) -> Optional[str]:
+@dataclass
+class TranscodeOutput:
+    transcoded_file_path: str
+    provenance: dict
+
+
+def try_transcode(input_path, asset_id, extension) -> Optional[TranscodeOutput]:
     logger.info(
         f"Determining if transcode is required for input_path: {input_path} asset_id: ({asset_id}) extension: ({extension})"
     )
+    start_time = time.time()
+
+    provenance = {
+        "activity_name": "Transcoding",
+        "activity_description": "Checks if input needs transcoding, then transcodes if so",
+        "processing_time_ms": 0,
+        "start_time_unix": start_time,
+        "parameters": [],
+        "software_version": "",
+        "input_data": input_path,
+        "output_data": "",
+        "steps": [],
+    }
 
     # if it's alrady valid audio no transcode necessary
     if _is_audio_file(extension):
         logger.info("No transcode required, input is audio")
-        return input_path
+        end_time = (time.time() - start_time) * 1000
+        provenance["processing_time_ms"] = end_time
+        provenance["output_data"] = input_path
+        provenance["steps"].append("No transcode required, input is audio")
```

Comment:
Here again we need to think (in the future) a bit more clearly as a team about how we want to communicate about steps that were skipped (but not failed).